content/en/blog/_posts/2023-04-11-volume-group-snapshot-alpha.md

```diff
@@ -0,0 +1,165 @@
+
+## How do I use Kubernetes Volume Group Snapshots
+
+Volume Group Snapshot feature is implemented in the external-snapshotter repo. It contains the following components:
+
+* Kubernetes Volume Group Snapshot CRDs
+* Volume group snapshot controller logic is added to the common snapshot controller.
+* Volume group snapshot validation webhook logic is added to the common snapshot validation webhook.
+* Logic to make CSI calls is added to CSI Snapshotter sidecar controller.
+
+The volume snapshot controller, CRDs, and validation webhook are deployed once per cluster, while the sidecar is bundled with each CSI driver.
+
+Therefore, it makes sense to deploy the volume snapshot controller, CRDs, and validation webhook as a cluster addon. It is strongly recommended that Kubernetes distributors bundle and deploy the volume snapshot controller, CRDs, and validation webhook as part of their Kubernetes cluster management process (independent of any CSI Driver).
+
+### Creating a new group snapshot with Kubernetes
+Once a VolumeGroupSnapshotClass object is defined and you have volumes you want to snapshot together, you may create a new group snapshot by creating a VolumeGroupSnapshot object.
+
+The source of the group snapshot specifies whether the underlying group snapshot should be dynamically created or if a pre-existing VolumeGroupSnapshotContent should be used. One of the following members in the source must be set.
+* Selector - Selector is a label query over persistent volume claims that are to be grouped together for snapshotting. This labelSelector will be used to match the label added to a PVC.
+* VolumeGroupSnapshotContentName - specifies the name of a pre-existing VolumeGroupSnapshotContent object representing an existing volume group snapshot.
+
+For dynamic provisioning, a selector must be set so that the snapshot controller can find PVCs with the matching labels to be snapshotted together.
+
+```
+apiVersion: groupsnapshot.storage.k8s.io/v1alpha1
+kind: VolumeGroupSnapshot
+metadata:
+  name: new-group-snapshot-demo
+  namespace: demo-namespace
+spec:
+  volumeGroupSnapshotClassName: csi-groupSnapclass
+  source:
+    selector:
+      group: myGroup
+```
+
+In the VolumeGroupSnapshot spec, a user can specify the VolumeGroupSnapshotClass which has the information about which CSI driver should be used for creating the group snapshot.
+
+### Importing an existing group snapshot with Kubernetes
+You can always import an existing group snapshot to Kubernetes by manually creating a VolumeGroupSnapshotContent object to represent the existing group snapshot. Because VolumeGroupSnapshotContent is a non-namespace API object, only a system admin may have the permission to create it. Once a VolumeGroupSnapshotContent object is created, the user can create a VolumeGroupSnapshot object pointing to the VolumeGroupSnapshotContent object.
+```
```

Comment:
You can mark this as YAML and get highlighting.

Reply:
Done